File: ambeth/Ambeth.Privilege/ambeth/transfer/PrivilegeResult.cs
using System;
using System.Collections.Generic;
using De.Osthus.Ambeth.Annotation;
using De.Osthus.Ambeth.Merge.Model;
using De.Osthus.Ambeth.Model;
using System.Runtime.Serialization;
using De.Osthus.Ambeth.Privilege.Model;

namespace De.Osthus.Ambeth.Privilege.Transfer
{
    [DataContract(Name = "PrivilegeResult", Namespace = "http://schemas.osthus.de/Ambeth")]
    public class PrivilegeResult : IPrivilegeResult
    {       

        [DataMember]
        public IObjRef Reference { get; set; }

        [DataMember]
        public ISecurityScope SecurityScope { get; set; }

        [DataMember]
        public PrivilegeEnum[] Privileges { get; set; }
    }
}

File: jambeth/jAmbeth.Privilege/src/main/java/de/osthus/ambeth/privilege/transfer/PrivilegeResult.java
package de.osthus.ambeth.privilege.transfer;

import javax.xml.bind.annotation.XmlAccessType;
import javax.xml.bind.annotation.XmlAccessorType;
import javax.xml.bind.annotation.XmlElement;
import javax.xml.bind.annotation.XmlRootElement;

import de.osthus.ambeth.merge.model.IObjRef;
import de.osthus.ambeth.model.ISecurityScope;
import de.osthus.ambeth.privilege.model.IPrivilegeResult;
import de.osthus.ambeth.privilege.model.PrivilegeEnum;

@XmlRootElement(name = "PrivilegeResult", namespace = "http://schemas.osthus.de/Ambeth")
@XmlAccessorType(XmlAccessType.FIELD)
public class PrivilegeResult implements IPrivilegeResult
{
	@XmlElement(required = true)
	protected IObjRef reference;

	@XmlElement(required = true)
	protected ISecurityScope securityScope;

	@XmlElement(required = true)
	protected PrivilegeEnum[] privileges;

	@Override
	public IObjRef getReference()
	{
		return reference;
	}

	public void setReference(IObjRef reference)
	{
		this.reference = reference;
	}

	@Override
	public ISecurityScope getSecurityScope()
	{
		return securityScope;
	}

	public void setSecurityScope(ISecurityScope securityScope)
	{
		this.securityScope = securityScope;
	}

	@Override
	public PrivilegeEnum[] getPrivileges()
	{
		return privileges;
	}

	public void setPrivileges(PrivilegeEnum[] privileges)
	{
		this.privileges = privileges;
	}
}
